### PR TITLE
Trip scheduling logic

### PR DESCRIPTION
**Pull request by jpn--**
The BayDAG pull request included a change in trip scheduling that alters the results of the model for certain corner cases.  The change is not detectable in the current CI test suite, which only tests a very limited number of households and does not find many corner-case errors.  (The upcoming refactoring includes infrastructure to rigorously test much larger subsets of models, which is how this was discovered.  For the MTC test model, the change impacts the departure time of approximately 0.7% of all simulated trips.)

This PR rolls back the model component to use the original logic by default, and allows users to upgrade to the revised logic by setting "logic_version: 2" in the model settings.  If no logic level is set explicitly, a warning is emitted that "logic_version: 1" is implied, and that a future version of ActivitySim may default to logic version 2.

### CODE REVIEW

File: activitysim/abm/models/trip_scheduling.py [hunk 1]
```diff
@@ -145,11 +170,18 @@ def update_tour_earliest(trips, outbound_choices):
...
+            tmp_trips["outbound"], tmp_trips["max_outbound_departure"]
+        )
+    elif logic_version > 1:
+        tmp_trips["earliest"] = np.where(
+            ~tmp_trips["outbound"] & ~tmp_trips["max_outbound_departure"].isna(),
+            tmp_trips["max_outbound_departure"],
+            tmp_trips["earliest"],
+        )
+    else:
+        raise ValueError(f"bad logic_version: {logic_version}")
```
Comment (dhensle) [inline activitysim/abm/models/trip_scheduling.py:184]: I think we need a check somewhere in here that will require logic_version == 2 if the scheduling mode is "relative", otherwise that mode will fail. Should mention this requirement in the documentation as well.

**Review by dhensle — CHANGES_REQUESTED** [review]
Code looks fine to me minus my one comment. A lot of changes to keep some "broken" functionality, but I understand the desire to keep the previous large-scale versions unchanged.

File: activitysim/abm/models/trip_scheduling.py (hunk 1 above)
Comment (jpn--) [reply activitysim/abm/models/trip_scheduling.py:184]: I've added that check to the code, and a note in the documentation.

**Comment by jpn--** [comment]
Thanks @dhensle.  I agree there's a lot of dancing around here to preserve a "broken" model, but this is the price we need to pay to keep everyone on one platform.

File: activitysim/abm/models/trip_scheduling.py (hunk 1 above)
Comment (dhensle) [reply activitysim/abm/models/trip_scheduling.py:184]: Looks good, thanks @jpn-- .  (And thanks for fixing my spelling errors in the documentation too!)
